doc/source/serve/deployment-graph.md

```diff
@@ -0,0 +1,565 @@
+```
+
++++
+
+### Step 2: Model and combiner class instantiation
+
+After we got the preprocessed inputs, we're ready to combine them to construct request object we want to sent to two models instantiated with different initial weights. This means we need:
+
+(1) Two `Model` instances in the graph instantiated with different initial weights
+<br>
+(2) A `Combiner` that refereces `Model` nodes for its runtime implementation by passing them as init args in `.bind()`
+<br>
+(3) The ability of `Combiner` to receive and merge preprocessed inputs for the same user input, even they might be produced async and received out of order.
+
++++
+
+```python
+# ... previous nodes implementation skipped
+
+@serve.deployment
+class Model:
+    def __init__(self, weight: int):
+        self.weight = weight
+
+    async def forward(self, input: int):
+        time.sleep(0.3) # Manual delay for blocking computation 
+        return f"({self.weight} * {input})"
+
+
+@serve.deployment
+class Combiner:
+    def __init__(self, m1: Model, m2: Model):
+        self.m1 = m1
+        self.m2 = m2
+
+    async def run(self, req_part_1, req_part_2, operation):
+        # Merge model input from two preprocessors  
+        req = f"({req_part_1} + {req_part_2})"
+        
+        # Submit to both m1 and m2 with same req data in parallel
+        r1_ref = self.m1.forward.remote(req)
+        r2_ref = self.m2.forward.remote(req)
+        
+        # Async gathering of model forward results for same request data
+        rst = await asyncio.gather(*[r1_ref, r2_ref])
+
+
+# DAG building
+with InputNode() as dag_input:
+    preprocessed_1 = preprocessor.bind(dag_input[0])  # Partial access of user input by index
+    preprocessed_2 = avg_preprocessor.bind(dag_input[1]) # Partial access of user input by index
+    m1 = Model.bind(1)
+    m2 = Model.bind(2)
+    combiner = Combiner.bind(m1, m2)
+    dag = combiner.run.bind(
+        preprocessed_1, preprocessed_2, dag_input[2]  # Partial access of user input by index
+    ) 
+```
+
++++
+
+We are adding a few more pieces to our dag builder: `bind()` on class and class method, as well as passing the output of `Model.bind()` as init args into another class `Combiner.bind()`
+
+#### bind() on class constructor 
+
+**`Class.bind(*args, **kwargs)`** constructs and returns a DAGNode that acts as the instantiated instance of Class, where `*args` and `**kwargs` are used as init args. In our implementation, we have
+
+```python
+m1 = Model.bind(1)
+m2 = Model.bind(2)
+```
+
+This means we're creating two more DAGNodes of an instance of `Model` that is constructed with init arg of `1` and `2`, and refereced with variable name `m1`, `m2` respectively.
```

Comment:
still think this should be combined with the part about functions, they're basically the same thing except classes take args

Reply:
moved, seem like it didn't hurt reading experience when we introduce passing deployment node as args